src/request/actions/ActionCreator.ts

```diff
@@ -0,0 +1,32 @@
+import { Action } from 'redux';
+
+interface IActionCreatorAction<TPayload> extends Action {
+    payload: TPayload;
+}
+
+interface IActionCreatorBase<TPayload> {
+    type: string;
+    unwrap(action: Action): TPayload;
+}
+
+interface IActionCreator<TPayload> extends IActionCreatorBase<TPayload> {
+    (payload: TPayload): IActionCreatorAction<TPayload>;
+}
+
+export function createActionCreator<TPayload>(type: string): IActionCreator<TPayload> {
+    const actionCreator = (payload: TPayload) => {
+        return {
+            type: type,
+            payload: payload
+        }
+    };
+    
+    const typedActionCreator = <IActionCreator<TPayload>>actionCreator;
+    
+    typedActionCreator.type = type;
+    typedActionCreator.unwrap = (action: Action) => {
+        return (<IActionCreatorAction<TPayload>>action).payload;
+    };
```

Comment:
I'm guessing the reason to have `unwrap` as apposed to just getting the payload property is because this method allows us to access the payload already cast?